src/perf/isolates.md

```diff
@@ -0,0 +1,286 @@
+runs the computation on the main thread on the web, but spawns a new thread on
+mobile devices. On mobile and desktop platforms `await compute(fun, message)` is
+equivalent to `await Isolate.run(() => fun(message))`.
+
+## Stateful, longer-lived isolates
+
+- [`compute()`][] abstracts a handful of lower-level, isolate-related APIs to  simplify isolate management:
+- [`Isolate.spawn()`][] and [`Isolate.exit()`][]
+- [`ReceivePort`][] and [`SendPort`][]
+- [`send()`][] method
+
+When you use the compute method, the new isolate immediately shuts down after it
+returns a single message to the main isolate. Sometimes, you’ll need isolates
+that are long lived, and can pass multiple messages to each other over time. In
+Dart, you can accomplish this with the Isolate API and Ports. These long-lived
+isolates are colloquially known as _background workers_.
+
+Long-lived isolates are useful when you have a specific process that either
+needs to be run repeatedly throughout the lifetime of your application, or if
+you have a process that runs over a period of time and needs to yield multiple
+return values to the main isolate.
+
+### ReceivePorts and SendPorts
+
+Set up long-lived communication between isolates with two classes (in addition
+to Isolate): [`ReceivePort`][] and [`SendPort`][]. These ports are the only way isolates can
+communicate with each other.
+
+Ports behave similarly to Streams, in which the StreamController or Sink is
+created in one isolate, and the listener is set up in the other isolate. In this
+analogy, the StreamConroller is called a SendPort, and you can “add” messages
+with the send() method. ReceivePorts are the listeners, and when these listeners
+receive a new message, they call a provided callback with the message as an
+argument.
+
+For an in-depth explanation on setting up two-way communication between the main
+isolate and a worker isolate, follow the examples in the [Dart documentation][].
+
+[Dart documentation]: {{site.dart-site}}/language/concurrency
+
+## Common use cases for isolates
+
+There is only one hard rule for when you should use isolates, and that’s when
+large computations are causing your Flutter application to experience UI jank.
+This jank happens when there is any computation that takes longer than the
+Flutter’s frame gap.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Any process _could_ take longer to complete, depending on the implementation and the input data, making it impossible to create an exhaustive list of when you need to consider using isolates.
+
+That said, isolates are commonly used for the following:
+
+- Reading data from a local database
+- Sending push notifications
+- Parsing and decoding large data files
+- Processing or compressing photos, audio files, and video files
+- Converting audio and video files
+- When you need asynchronous support while using FFI
+- Applying filtering to complex lists or filesystems
```

Comment:
I can't help but think that this very useful list should appear MUCH higher in this page.